src/ConstructiveSolidGeometry/VolumePrimitives/VolumePrimitives.jl

```diff
@@ -1,5 +1,19 @@
-include("Cone.jl")
+ClosedPrimitive(p::VP) where {VP <:AbstractVolumePrimitive} = VP(p, COT = ClosedPrimitive)
+OpenPrimitive(p::VP) where {VP <: AbstractVolumePrimitive} = VP(p, COT = OpenPrimitive)
+switchClosedOpen(p::AbstractVolumePrimitive{<:Any,ClosedPrimitive}) = OpenPrimitive(p)
+switchClosedOpen(p::AbstractVolumePrimitive{<:Any,OpenPrimitive}) = ClosedPrimitive(p)
+
+distance(pt::CartesianPoint, vp::AbstractVolumePrimitive) = 
+minimum(map(p -> distance(pt, p), surfaces(vp)))
+
+function sample(vp::AbstractVolumePrimitive) 
+    surfs = surfaces(vp)
+    ls = unique!(collect(Base.Iterators.flatten(lines.(surfs))))
+    unique!(mapreduce(sample, vcat, ls))
```

Comment:
Does this sample everything with the default number of points?

Reply:
Currently, yes. But, we can add some methods to tune this. 
The creation of the initial grid has to be improved. 
There we will need it. But not in this PR.